browser/ui/views/brave_help_bubble/brave_help_bubble_host_view.cc

```diff
@@ -0,0 +1,230 @@
+#include "third_party/skia/include/core/SkColor.h"
+#include "third_party/skia/include/core/SkPoint.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/compositor/layer.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/geometry/transform_util.h"
+#include "ui/views/animation/animation_builder.h"
+#include "ui/views/background.h"
+#include "ui/views/bubble/bubble_border.h"
+#include "ui/views/bubble/bubble_dialog_delegate_view.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/layout/fill_layout.h"
+#include "ui/views/view_class_properties.h"
+
+namespace {
+constexpr int kWidth = 60;
+constexpr int kHeight = 60;
+constexpr base::TimeDelta kPulsingDuration = base::Milliseconds(1000);
+constexpr SkColor kBlockColor = SkColorSetRGB(52, 172, 224);
+constexpr SkPoint kPts[] = {{0, 0}, {60, 60}};
+constexpr SkColor4f kColors[] = {{0.66, 0.10, 0.47, 1.f},
+                                 {0.22, 0.17, 0.81, 1.f}};
+constexpr SkScalar kPositions[] = {0.0, 0.65, 1.0};
+sk_sp<cc::PaintShader> kBraveGradient =
+    cc::PaintShader::MakeLinearGradient(kPts,
+                                        kColors,
+                                        kPositions,
+                                        3,
+                                        SkTileMode::kClamp);
+
+void SchedulePulsingAnimation(ui::Layer* layer) {
+  DCHECK(layer);
+
+  const gfx::Rect local_bounds(layer->bounds().size());
+  views::AnimationBuilder()
+      .SetPreemptionStrategy(
+          ui::LayerAnimator::IMMEDIATELY_ANIMATE_TO_NEW_TARGET)
+      .Repeatedly()
+      .SetDuration(kPulsingDuration)
+      .SetTransform(layer,
+                    gfx::GetScaleTransform(local_bounds.CenterPoint(), 0.7f),
+                    gfx::Tween::EASE_IN)
+      .At(kPulsingDuration)
+      .SetDuration(kPulsingDuration)
+      .SetTransform(layer,
+                    gfx::GetScaleTransform(local_bounds.CenterPoint(), 1.0f),
+                    gfx::Tween::EASE_OUT);
+}
+}  // namespace
+
+BraveHelpBubbleHostView::BraveHelpBubbleHostView(View* tracked_element)
+    : tracked_element_(tracked_element) {
+  // Making the layer of this view transparent for mouse clicks to interact with
+  // the underlying tracked element
+  SetEnabled(false);
+}
+
+BraveHelpBubbleHostView::~BraveHelpBubbleHostView() {
+  if (brave_help_bubble_delegate_view_) {
+    brave_help_bubble_delegate_view_->RemoveObserver(this);
+  }
+
+  scoped_observation_.Reset();
+}
+
+// static
+base::WeakPtr<BraveHelpBubbleHostView> BraveHelpBubbleHostView::Create(
+    View* tracked_element,
+    const std::string text) {
+  auto* bubble = new BraveHelpBubbleHostView(tracked_element);
+  bubble->text_ = text;
+  return bubble->weak_factory_.GetWeakPtr();
+}
+
+void BraveHelpBubbleHostView::Show() {
+  if (!brave_help_bubble_delegate_view_) {
+    brave_help_bubble_delegate_view_ =
+        new BraveHelpBubbleDelegateView(this, text_);
+    brave_help_bubble_delegate_view_->AddObserver(this);
```

Comment:
nice, I never thought of this.

So, will this remove the code where we're removing the observer? Since `base::ScopedObservation` manages it for us?

Reply:
> Since base::ScopedObservation manages it for us?

Yeah, you don't need to call it from destructor. But if there were calls from other places, you might still need to call ScopedObservation's Reset().